### PR TITLE
Ganglion Impedance Widget - Dec 2021 Update

### LINKED ISSUE #1021: Revisit Ganglion Impedance Check Widget
## Problem

Toggles data stream and saves impedance data to recording file. Also, triggers lost packet messages in console log.

## Expected 

Should not toggle the "Start Data Stream" button. Behavior should be similar to Cyton signal check widget. Impedance data should not be streamed using BrainFlow streamer or saved to file in the GUI.

About packetlosstracker, maybe just set this to `null` and then re-instantiate it when toggling Ganglion impedance check.

## Operating System and Version

All OS

## GUI Version

5.1.0-alpha.1

## Running standalone app

Developer copy

## Type of OpenBCI Board

Ganglion

## Are you using a WiFi Shield?

Both! Ganglion and Ganglion+WiFi

## Console Log
[Console_2021-12-08_15-35-44.txt](https://github.com/OpenBCI/OpenBCI_GUI/files/7679346/Console_2021-12-08_15-35-44.txt)

[Brainflow_2021-12-08_15-36-46.txt](https://github.com/OpenBCI/OpenBCI_GUI/files/7679350/Brainflow_2021-12-08_15-36-46.txt)

```
CP: Playback History Size = 3
wifiList: click! 
[DEFAULT]: Selected WiFi Board: OpenBCI-E476, WiFi IP Address: 10.0.1.40
[DEFAULT]: Attempting to Start Session...
initButtonPressed: Calling initSystem() after next draw()
OpenBCI_GUI: Start session. Calling initSystem().


=================================================
||             INITIALIZING SYSTEM             ||
=================================================

Sending config string to board: n
Sending config string to board: ~7
OpenBCI_GUI: Channel count set to 4
InitSettings: Saving Default Settings to file!
SessionSettings: 4 active widgets saved!
[SUCCESS]: Session started!
headPlot: computeWeightFactor: Ielec 0, solution complete with 2000 iterations. max_dVal = 1.0759383E-5
headPlot: computeWeightFactor: Ielec 1, solution complete with 2000 iterations. max_dVal = 1.0493398E-5
headPlot: computeWeightFactor: Ielec 2, solution complete with 2000 iterations. max_dVal = 8.711815E-5
headPlot: computeWeightFactor: Ielec 3, solution complete with 2000 iterations. max_dVal = 8.860826E-5
[DEFAULT]: openBCI_GUI: startButton was pressed. Starting data transfer, wait a few seconds.
OpenBCI_GUI: openNewLogFile: opened ODF output file: /Users/Richard/Documents/OpenBCI_GUI/Recordings/OpenBCISession_2021-12-08_15-36-44/OpenBCI-RAW-2021-12-08_15-37-02.txt
OpenBCI_GUI: BrainFlow Streamer Location: file:///Users/Richard/Documents/OpenBCI_GUI/Recordings/OpenBCISession_2021-12-08_15-36-44//BrainFlow-RAW_2021-12-08_15-36-44_0.csv:w
Brainflow start streaming
[DEFAULT]: Data stream started.
Starting Ganglion impedance check...
Brainflow stop streaming
[DEFAULT]: Data stream stopped.
Sending config string to board: z
Brainflow start streaming
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
Brainflow stop streaming
Sending config string to board: Z
[DEFAULT]: openBCI_GUI: startButton was pressed. Starting data transfer, wait a few seconds.
OpenBCI_GUI: openNewLogFile: opened ODF output file: /Users/Richard/Documents/OpenBCI_GUI/Recordings/OpenBCISession_2021-12-08_15-36-44/OpenBCI-RAW-2021-12-08_15-37-30.txt
OpenBCI_GUI: BrainFlow Streamer Location: file:///Users/Richard/Documents/OpenBCI_GUI/Recordings/OpenBCISession_2021-12-08_15-36-44//BrainFlow-RAW_2021-12-08_15-36-44_2.csv:w
Brainflow start streaming
ERROR: Exception when starting stream
brainflow.BrainFlowError: Error in start_stream:BOARD_WRITE_ERROR
	at brainflow.BoardShim.start_stream(BoardShim.java:644)
	at OpenBCI_GUI$BoardBrainFlow.startStreaming(OpenBCI_GUI.java:2746)
	at OpenBCI_GUI.startRunning(OpenBCI_GUI.java:867)
	at OpenBCI_GUI$TopNav.stopButtonWasPressed(OpenBCI_GUI.java:16941)
	at OpenBCI_GUI$TopNav$2.controlEvent(OpenBCI_GUI.java:16773)
	at controlP5.Controller.callListener(Unknown Source)
	at controlP5.Controller.setMousePressed(Unknown Source)
	at controlP5.ControllerGroup.setMousePressed(Unknown Source)
	at controlP5.ControlWindow.mouseReleasedEvent(Unknown Source)
	at controlP5.ControlWindow.mouseEvent(Unknown Source)
	at jdk.internal.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at processing.core.PApplet$RegisteredMethods.handle(PApplet.java:1387)
	at processing.core.PApplet.handleMethods(PApplet.java:1541)
	at processing.core.PApplet.handleMouseEvent(PApplet.java:2484)
	at processing.core.PApplet.dequeueEvents(PApplet.java:2374)
	at processing.core.PApplet.handleDraw(PApplet.java:2212)
	at processing.opengl.PSurfaceJOGL$DrawListener.display(PSurfaceJOGL.java:894)
	at jogamp.opengl.GLDrawableHelper.displayImpl(GLDrawableHelper.java:692)
	at jogamp.opengl.GLDrawableHelper.display(GLDrawableHelper.java:674)
	at jogamp.opengl.GLAutoDrawableBase$2.run(GLAutoDrawableBase.java:443)
	at jogamp.opengl.GLDrawableHelper.invokeGLImpl(GLDrawableHelper.java:1293)
	at jogamp.opengl.GLDrawableHelper.invokeGL(GLDrawableHelper.java:1147)
	at com.jogamp.newt.opengl.GLWindow.display(GLWindow.java:782)
	at com.jogamp.opengl.util.AWTAnimatorImpl.display(AWTAnimatorImpl.java:81)
	at com.jogamp.opengl.util.AnimatorBase.display(AnimatorBase.java:453)
	at com.jogamp.opengl.util.FPSAnimator$MainTask.run(FPSAnimator.java:178)
	at java.base/java.util.TimerThread.mainLoop(Timer.java:556)
	at java.base/java.util.TimerThread.run(Timer.java:506)
[ERROR]: Failed to start data stream. Please check hardware. See Console Log or BrainFlow Log for more details.
[DEFAULT]: openBCI_GUI: startButton was pressed. Starting data transfer, wait a few seconds.
OpenBCI_GUI: BrainFlow Streamer Location: file:///Users/Richard/Documents/OpenBCI_GUI/Recordings/OpenBCISession_2021-12-08_15-36-44//BrainFlow-RAW_2021-12-08_15-36-44_2.csv:w
Brainflow start streaming
[DEFAULT]: Data stream started.
[DEFAULT]: openBCI_GUI: stopButton was pressed. Stopping data transfer, wait a few seconds.
Brainflow stop streaming
[DEFAULT]: Data stream stopped.
Starting Ganglion impedance check...
Sending config string to board: z
Brainflow start streaming
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
WARNING: Lost 200 Samples Between 0-0
OpenBCI_GUI: mousePressed: clicked in CP box
openBCI_GUI: haltSystem: Halting system for reconfiguration of settings...
[DEFAULT]: Data stream is already stopped.
CP: Playback History Size = 3
SHUTDOWN HOOK
```

### PR DESCRIPTION
**Pull request by retiutut**
Fixes #1021

### CODE REVIEW

**Review by Andrey1994 — APPROVED** [review]

File: OpenBCI_GUI/W_GanglionImpedance.pde [hunk 1]
```diff
@@ -103,7 +103,6 @@ class W_GanglionImpedance extends Widget {
                     if (!ganglionBoard.isCheckingImpedance()) {
                         // We need to either stop the time series data, or allow it to scroll, like currently. 
                         // the values in time series are not meaningful when Impedance check is active
-                        topNav.stopButtonWasPressed();
```
Comment (Andrey1994) [inline OpenBCI_GUI/W_GanglionImpedance.pde:106]: are you sure about this? there is a comment which explains why we stop it... at least lets remove this comment also
And I remember we had issues with impedance for ganglion, I am a little scared to touch this

**Review by Andrey1994 — CHANGES_REQUESTED** [review]

File: OpenBCI_GUI/W_GanglionImpedance.pde (hunk 1 above)
Comment (retiutut) [reply OpenBCI_GUI/W_GanglionImpedance.pde:106]: @Andrey1994 https://github.com/OpenBCI/OpenBCI_GUI/issues/1021

It does bad things for the App and PacketLossTracker.

Comment (retiutut) [reply OpenBCI_GUI/W_GanglionImpedance.pde:106]: I can review this PR again before merging.

**Comment by retiutut** [comment]
This PR addresses the issue and the GUI behaves properly.

**Pull request merged**